samples/nrf9160/lwm2m_client/prj-ci.conf

```diff
@@ -0,0 +1,112 @@
+# General config
+CONFIG_NEWLIB_LIBC=y
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_ASSERT=y
+CONFIG_REBOOT=y
+
+# Network
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_OFFLOAD=y
+
+# LwM2M
+CONFIG_LWM2M=y
+CONFIG_LWM2M_RW_JSON_SUPPORT=n
+CONFIG_LWM2M_SERVER_DEFAULT_PMIN=1
+CONFIG_LWM2M_SERVER_DEFAULT_PMAX=300
+CONFIG_LWM2M_CONN_MON_OBJ_SUPPORT=y
+CONFIG_LWM2M_CONN_MON_BEARER_MAX=2
+CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT=y
+CONFIG_LWM2M_LOCATION_OBJ_SUPPORT=y
+CONFIG_LWM2M_IPSO_SUPPORT=y
+CONFIG_LWM2M_IPSO_TEMP_SENSOR=y
+CONFIG_LWM2M_IPSO_LIGHT_CONTROL=y
+CONFIG_LWM2M_IPSO_BUZZER=y
+CONFIG_LWM2M_IPSO_ONOFF_SWITCH=y
+CONFIG_LWM2M_IPSO_ONOFF_SWITCH_INSTANCE_COUNT=2
+CONFIG_LWM2M_IPSO_PUSH_BUTTON=y
+CONFIG_LWM2M_IPSO_PUSH_BUTTON_INSTANCE_COUNT=2
+CONFIG_LWM2M_IPSO_ACCELEROMETER=y
+
+# DTLS settings
+CONFIG_LWM2M_DTLS_SUPPORT=y
+CONFIG_LWM2M_PEER_PORT=5684
+
+# MBEDTLS settings
+CONFIG_MBEDTLS_CIPHER_CCM_ENABLED=y
+# Disable RSA, we don't parse certs: saves flash/memory
+CONFIG_MBEDTLS_KEY_EXCHANGE_RSA_ENABLED=n
+# Enable PSK instead
+CONFIG_MBEDTLS_KEY_EXCHANGE_PSK_ENABLED=y
+CONFIG_NET_SOCKETS_TLS_MAX_CONTEXTS=4
+
+# Default app to debug logging
+CONFIG_LOG=y
+CONFIG_APP_LOG_LEVEL_DBG=y
+
+# Expand logging memory to avoid log_strdup() errors
+CONFIG_LOG_STRDUP_MAX_STRING=64
+CONFIG_LOG_STRDUP_BUF_COUNT=32
+
+# Support HEX style PSK values (double the size + NULL char)
+CONFIG_LWM2M_SECURITY_KEY_SIZE=33
+
+# extend CoAP retry timing to 10 seconds for LTE/LTE-M
+CONFIG_COAP_INIT_ACK_TIMEOUT_MS=10000
+
+# UI settings
+CONFIG_UI_BUTTON=y
+CONFIG_UI_BUZZER=y
+
+# FOTA
+CONFIG_BOOTLOADER_MCUBOOT=y
+CONFIG_IMG_MANAGER=y
+CONFIG_MCUBOOT_IMG_MANAGER=y
+CONFIG_IMG_ERASE_PROGRESSIVELY=y
+CONFIG_FLASH=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+# Enable temp sensor
+CONFIG_SENSOR=y
+# TODO: No DIE temp sensor on nRF9150, select a temp sensor?
+# CONFIG_TEMP_NRF5=y
+
+# Enable settings storage
+CONFIG_SETTINGS=y
+CONFIG_FCB=y
+CONFIG_SETTINGS_FCB=y
+
+# LTE link control
+CONFIG_LTE_LINK_CONTROL=y
+CONFIG_LTE_AUTO_INIT_AND_CONNECT=n
+
+# BSD library
+CONFIG_BSD_LIBRARY=y
+
+# Modem info
+CONFIG_MODEM_INFO=y
+
+# Library for buttons and LEDs
+CONFIG_DK_LIBRARY=y
+CONFIG_DK_LIBRARY_INVERT_LEDS=n
+
+# Enable shell
+CONFIG_NET_SHELL=y
+
+# Heap and stacks
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+CONFIG_MAIN_STACK_SIZE=8192
+CONFIG_GPS_SIM_THREAD_STACK_SIZE=1024
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
+#CONFIG_HW_STACK_PROTECTION=y
+
+# Disable native network stack to save some memory
+CONFIG_NET_IPV4=n
+CONFIG_NET_IPV6=n
+CONFIG_NET_UDP=n
+CONFIG_NET_TCP=n
+CONFIG_NET_RX_STACK_SIZE=256
+CONFIG_NET_TX_STACK_SIZE=256
+
+# Set LwM2M Server IP address here
+CONFIG_APP_LWM2M_SERVER="passci"
```

Comment:
How about `overlay-ci.conf` contatining:
```
# Set LwM2M Server IP address here
CONFIG_APP_LWM2M_SERVER="passci"
```
only? There's a risk that `prj.conf` && `prj-ci.conf` will diverse over time, and the CI will not build the actual sample configuration. You can apply the overlay like this:
`CONF_FILE="prj.conf overlay-ci.conf"`
Should also be possible with `OVERLAY_CONFGI=overlay-ci.conf`, though it had issues with multiimage build (fix was merged today into fw-nrfconnect-zephyr).

Reply:
You are correct, I was also nervous about separate conf file.  I will try overlay fix again today and send an updated patch (including zephyr SHA change).